### DISCUSSION TITLE
keyball61 rev2 右側トラックボール、左側のキー反応およびランド剥離について

### DISCUSSION THREAD

**Discussion by CheapHead** [Yowkees/keyball#361] [Q&A] [answered]
Yowkees様

表題の件についてアドバイスいただけないでしょうか。

購入モデル：keyball 61 rev2
購入先：遊舎工房
右手：トラックボール
左手：標準

症状：①キー６，７，８，９(列全部)、２３、ロープロ両サイドキー反応せず
　　　➁２３ランド剥離

詳細：ビルドガイドに従って組み立てたつもりなのですが、
写真の赤枠で囲った6～9(1列全部)、23、ロープロファイル両側(左、右)がキー反応しないです。
ファームウェアは本番用で試しました。
OLEDは接続時に左右両方表示されます。また、ダイオードの向きも確認しています。
キーソケットの不具合を疑って何度かハンダを付けなおして試したのですが、
症状は変わらずの状態で結局いったん取り外したところ、23のランドまで取れてしまい、
ほとほと困ってしまいました。
![基盤左側_裏](https://github.com/Yowkees/keyball/assets/34470444/67360c7e-33fb-4eae-89f0-9f3690621f9f)

ダイオードの向きとキーソケットの接続不良以外に疑うべきところはないと思うのですが、
ほかに何か確認するべき個所としてアドバイスいただけないでしょうか。
また、23のランド剥離ですが、別の問い合わせでリード線でどれかとどれかをつなぐことで回避できるような
回答がありましたが、こちらでも同様の対応はできないでしょうか。
一生使うキーボードと決め込んで購入したのでここであきらめたくないです。
どうかアドバイスをお願いいたします。
追記:上記のキーはピンセットなどでショートさせても反応がありませんでした。

**Comment by Yowkees** [comment, marked as answer]
ご連絡ありがとうございます。状況了解しました。

まず縦一列反応していないものは、下記画像の青丸のダイオードがテンプラ（半田付け不良）しているように見えます。このダイオードがテンプラしていると縦一列入力できなくなるのでご確認ください。

次に左下のロープロキーですが、キーソケットの緑丸をつけた部分の半田付けが怪しい気がしました。半田が基板パッドまで届いているでしょうか。もしくは同じく緑丸を付けたダイオードがこのキーに該当するものですので、半田付けご確認お願いします。

最後に23のキーですが、黄色丸をつけたダイオードの左側の半田ボールが大きく、テンプラしている可能性があるかと思います。
また、パッドが剥がれてしまったということで、オレンジ点どうしをエナメル線かゴム被覆導線で接続していただくと復活できます。
ただ、キーソケットの保持力が無い状態ですので、キーソケットを両面テープ等で基板に固定していただくと良いと思います。

以上、ぜひ末永くご使用いただきたいのでお手数ですがご確認お願いいたします。

![KB61_keyfix](https://github.com/Yowkees/keyball/assets/84209772/0769353b-9736-42f3-bfd9-87fe7fb12a74)

**Reply by CheapHead** [reply]
Yowkees様
アドバイスありがとうございます。無事すべてのキーが反応するようになりました。
初めての自作でここまでできて非常にうれしいです。
どうもありがとうございます。

厚かましいのは重々承知でもう一つお伺いさせてください。
組み上げてファームウェアを書き込みRemapで認識させると、
１．右手トラックボールあり側ProMicroに接続すると、左右ともにトラックボールなしで認識
２．左手トラックボールなし側ProMicroに接続すると、左右ともに正しくキー配列を認識
となってしまいます。

左手側のLEFTは正しくショートさせていると思うのですが、何がいけないのでしょうか。
![LeffSide](https://github.com/Yowkees/keyball/assets/34470444/ac515748-5a99-4766-a504-757eff6a2c61)
追記:trrsで左右を繋いでProMicroのDATA端子間の導通は確認しました
追記:右側で接続した時はREMAPでは両サイドトラックボールなしの認識でした。
![IMG_8491](https://github.com/Yowkees/keyball/assets/34470444/d37953b5-848f-40fe-8685-ad415360287e)

**Comment by Yowkees** [comment]
キーが無事に反応するようになって、ひとまず良かったです。

REMAPでの自動判定に問題があるようですが、右側にUSBを接続した際にトラックボールは動きますか？

**Reply by CheapHead** [reply]
Yowkees様
返信ありがとうございます。
TRRSケーブルを交換したらあっさり認識してくれました。vanauxというところのを使っていたのですが、そちらと相性が悪かったようです。お騒がせしてすみませんでした。
問題になっていたことがすべて解決できました。
故障などの不具合でまた質問させていただくかもしれませんが、よろしくお願いします。
これでRemapのキーアサインに勧めます。
素晴らしい製品をありがとうございます。
